x/authority/keeper/msg_server_update_chain_info.go

```diff
@@ -24,8 +25,27 @@ func (k msgServer) UpdateChainInfo(
 	if err != nil {
 		return nil, errors.Wrap(types.ErrUnauthorized, err.Error())
 	}
-	// set chain info
-	k.SetChainInfo(ctx, msg.ChainInfo)
 
+	chainInfo := types.ChainInfo{}
+	updatedChainInfo := false
```

Comment:
```suggestion
	chainInfoExist := false
```
More descriptive imo as it is updated in any case